mocca-client/src/main/java/com/paypal/mocca/client/MoccaUtils.java

```diff
@@ -36,21 +43,60 @@ String getValue() {
     }
 
     /*
-     * Returns an optional with the Type parametrized inside the given type,
+     * Returns an optional containing the Type parameterized inside the given type,
      * if the given type is parameterized and equals to the outer reference type.
      * If it is not, an empty optional is returned.
      */
-    static Optional<Type> getInnerType(final Type type, final Class<?> outerTypeReference) {
-        if (!(type instanceof ParameterizedType)) return Optional.empty();
+    static Optional<Type> getInnerType(final Type type, final Type outerTypeReference) {
+        if (!isParameterizedType(type)) return Optional.empty();
 
         ParameterizedType parameterizedType = (ParameterizedType) type;
-        if (parameterizedType.getRawType().getTypeName().equals(outerTypeReference.getName())) {
+        if (parameterizedType.getRawType().getTypeName().equals(outerTypeReference.getTypeName())) {
             return Optional.of(parameterizedType.getActualTypeArguments()[0]);
         } else {
             return Optional.empty();
         }
     }
 
+    /*
+     * Returns the first parametrized type inside the given type.
+     * If the given type is not parameterized an IllegalArgumentException is thrown.
+     */
+    static Type getInnerType(final Type type) {
```

Comment:
I wish we had a little more organization of the 'utilities'.  I think this could have been called `Reflection` and `OperationType` pulled out of here.  No biggie, absolutely opinion.  I just have a long-standing hate of `XyzUtils` classes:)

More concerning is why don't we unit test these?  I'm assuming they are getting tested indirectly.  My preference would be specific tests for these.

Reply:
Done